src/python/turicreate/toolkits/image_classifier/image_classifier.py

```diff
@@ -250,29 +251,74 @@ def create(
         raise TypeError("Unrecognized value for 'validation_set'.")
 
     if feature is None:
-        feature = _tkutl._find_only_image_column(dataset)
+        try:
+            feature = image_analysis.find_only_image_extracted_features_column(dataset, model)
+            feature_type = "extracted_features_array"
+        except:
+            feature = None
+
+        if feature is None:
+            try:
+                feature = _tkutl._find_only_image_column(dataset)
+                feature_type = "image"
+            except:
```

Comment:
You shouldn't need this `except` block. `_find_only_image_column` will raise it's own exception, if it doesn't find what it needs.

Reply:
You are right, will remove it.